crypto/ex_data.c

```diff
@@ -358,11 +358,11 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         if (storage[i] && storage[i]->dup_func)
             storage[i]->dup_func(to, from, &ptr, i,
                                  storage[i]->argl, storage[i]->argp);
-        CRYPTO_set_ex_data(to, i, ptr);
+        ret &= CRYPTO_set_ex_data(to, i, ptr);
```

Comment:
You're right , the fix seems larger than just this point.
It deserve a separated PR to address it.
I will put this change in it.